shell/gpu/gpu_surface_metal.mm

```diff
@@ -93,65 +95,52 @@
     return nullptr;
   }
 
-  const auto scale = layer_.get().contentsScale;
-
-  auto next_drawable = fml::scoped_nsprotocol<id<CAMetalDrawable>>([[layer_ nextDrawable] retain]);
-  if (!next_drawable) {
-    FML_LOG(ERROR) << "Could not acquire next metal drawable.";
-    return nullptr;
+  // Get ready for the next frame no matter what.
+  if (next_drawable_ != nullptr) {
+    CFRelease(next_drawable_);
+    next_drawable_ = nullptr;
   }
 
-  auto metal_texture = fml::scoped_nsprotocol<id<MTLTexture>>([next_drawable.get().texture retain]);
-  if (!metal_texture) {
-    FML_LOG(ERROR) << "Could not acquire metal texture from drawable.";
-    return nullptr;
-  }
-
-  GrMtlTextureInfo metal_texture_info;
-  metal_texture_info.fTexture.reset(SkCFSafeRetain(metal_texture.get()));
-
-  GrBackendRenderTarget metal_render_target(bounds.width * scale,   // width
-                                            bounds.height * scale,  // height
-                                            1,                      // sample count
-                                            metal_texture_info      // metal texture info
+  auto surface = SkSurface::MakeFromCAMetalLayer(context_.get(),            // context
```

Comment:
Does next_drawable_ get set synchronously here?

Reply:
No. This will defer the next drawable acquisition till much later. In fact, it is almost guaranteed to not be set after this call. I would have captured it in one of the scoped wrappers otherwise.

Reply:
What happens if the `next_drawable_` isn't set until after we check if it's null in the callback?

Reply:
The next  `GPUSurfaceMetal::AcquireFrame` will collect it if there is a next frame. `GPUSurfaceMetal::~GPUSurfaceMetal` will collect it via `ReleaseUnusedDrawableIfNecessary` if there isn't.

Reply:
I should mention that either case is extremely unlikely because the surface frame itself is collected immediately after submission.